### PR TITLE
Don't use port 80 on nginx service if using https

### PR DESCRIPTION
**Pull request by hrosenbe**
Changes in this pull request:
- Release 2.0.5 removed port 80 from the nginx service, but there was an issue with the handling of the `ssl` parameter which is supposed to be able to turn off the use of https, in which case port 80 is needed.  The fix also introduced a spurious error message.  This pull request removes port 80 from the nginx service only if we are using https to communicate from the workload driver.

### CODE REVIEW

File: runHarness/AppInstance/AuctionKubernetesAppInstance.pm [hunk 1]
```diff
@@ -155,7 +155,11 @@ override 'getEdgeAddrsRef' => sub {
 	  	 $logger->error("There are no IP addresses for the Kubernetes nodes");
 		 exit 1;
 	  }	
-      my $httpPort = $cluster->kubernetesGetNodePortForPortNumber("app=auction,type=webServer", 80, $self->namespace);
+	  # The http port will only be used if the ssl parameter is true
+      my $httpPort = 80;
+      if (!$self->getParamValue('ssl')) {
+      	$cluster->kubernetesGetNodePortForPortNumber("app=auction,type=webServer", 80, $self->namespace);
```
Comment (bhoflich) [inline runHarness/AppInstance/AuctionKubernetesAppInstance.pm:161]: The return value can be ignored and just 80 used?

File: runHarness/AppInstance/AuctionKubernetesAppInstance.pm [hunk 2]
```diff
@@ -155,7 +155,11 @@ override 'getEdgeAddrsRef' => sub {
 	  	 $logger->error("There are no IP addresses for the Kubernetes nodes");
 		 exit 1;
 	  }	
-      my $httpPort = $cluster->kubernetesGetNodePortForPortNumber("app=auction,type=webServer", 80, $self->namespace);
+	  # The http port will only be used if the ssl parameter is true
```
Comment (bhoflich) [inline runHarness/AppInstance/AuctionKubernetesAppInstance.pm:158]: if the ssl parameter is false?

Comment (hrosenbe) [reply runHarness/AppInstance/AuctionKubernetesAppInstance.pm:158]: Good point.  in that case we should remove port 443

File: runHarness/AppInstance/AuctionKubernetesAppInstance.pm (hunk 1 above)
Comment (hrosenbe) [reply runHarness/AppInstance/AuctionKubernetesAppInstance.pm:161]: No.  This is a bug.

File: runHarness/AppInstance/AuctionKubernetesAppInstance.pm [hunk 3]
```diff
@@ -155,12 +155,16 @@ override 'getEdgeAddrsRef' => sub {
...
+	  # The http port will only be used if the ssl parameter is false
       my $httpPort = 80;
       if (!$self->getParamValue('ssl')) {
-      	$cluster->kubernetesGetNodePortForPortNumber("app=auction,type=webServer", 80, $self->namespace);
+      	$httpPort = $cluster->kubernetesGetNodePortForPortNumber("app=auction,type=webServer", 80, $self->namespace);
+      }
+	  # The https port will only be used if the ssl parameter is true
+	  my $httpsPort = 443;
+      if ($self->getParamValue('ssl')) {
+      	$httpPort = $cluster->kubernetesGetNodePortForPortNumber("app=auction,type=webServer", 443, $self->namespace);
```
Comment (bhoflich) [inline runHarness/AppInstance/AuctionKubernetesAppInstance.pm:166]: Should it be $httpsPort =  (with the s)?

Comment (hrosenbe) [reply runHarness/AppInstance/AuctionKubernetesAppInstance.pm:166]: yes

**Pull request merged**